### PR TITLE
CWG Poll 12: P2113R0 Proposed resolution for 2019 comment CA 112

### LINKED ISSUE #3696: [2020-02 CWG Poll 12] P2113R0 Proposed resolution for 2019 comment CA 112
http://wiki.edg.com/pub/Wg21prague/StrawPolls/p2113r0.html

### PR DESCRIPTION
**Pull request by jensmaurer**
Also fixes NB CA 112 (C++20 CD).

Fixes #3696.
Fixes cplusplus/nbballot#111.
Fixes cplusplus/nbballot#119.

### CODE REVIEW

File: source/templates.tex [hunk 1]
```diff
@@ -3887,6 +3889,79 @@
...
+are not deducible from their function parameters,
+neither template is more specialized than the other.
+\item
+If there is either no reordering or more than one reordering
+of the associated \grammarterm{template-parameter-list}
+such that the corresponding \grammarterm{template-parameter}{s}
+of the \grammarterm{template-parameter-list}{s} are equivalent and
+such that the function parameters that positionally correspond
+between the two templates are of the same type,
+neither template is more specialized than the other.
```
Comment (burblebee) [inline source/templates.tex]: Ack!  Please add a separate editorial commit which breaks this horrendous sentence up via subbullets.

File: source/templates.tex [hunk 2]
```diff
@@ -3887,6 +3889,79 @@
...
 \end{example}
 
+\pnum
+If deduction against the other template succeeds for both transformed templates,
+constraints can be considered as follows:
+\begin{itemize}
+\item
+If their \grammarterm{template-parameter-list}{s}
+(possibly including \grammarterm{template-parameter}{s}
+invented for an abbreviated function template\iref{dcl.fct}) or
```
Comment (burblebee) [inline source/templates.tex]: delete ")" after reference

File: source/templates.tex [hunk 3]
```diff
@@ -3887,6 +3889,79 @@
...
+such that the corresponding \grammarterm{template-parameter}{s}
+of the \grammarterm{template-parameter-list}{s} are equivalent and
+such that the function parameters that positionally correspond
+between the two templates are of the same type,
+neither template is more specialized than the other.
+\end{itemize}
+\item
+Otherwise, if the corresponding \grammarterm{template-parameter}{s}
+of the \grammarterm{template-parameter-list}{s}
+are not equivalent\iref{temp.over.link}) or
```
Comment (burblebee) [inline source/templates.tex]: delete ")" after reference

File: source/templates.tex [hunk 4]
```diff
@@ -3887,6 +3889,79 @@
...
+  f(*ap, *bp);                  // OK: Can use different methods to produce template parameters
+}
+
+template <typename T, typename U> struct X {};
+
+template <typename T, C U, typename V> bool operator==(X<T, U>, V) = delete;
+template <C T, C U, C V>               bool operator==(T, X<U, V>);
+
+void h() {
+  X<void *, int>{} == 0;        // OK: Correspondence of [T, U, V] and [U, V, T]
```
Comment (burblebee) [inline source/templates.tex]: encode code in comment with \tcode

File: source/templates.tex (hunk 2 above)
Comment (jensmaurer) [reply source/templates.tex]: Fixed.

File: source/templates.tex (hunk 4 above)
Comment (jensmaurer) [reply source/templates.tex]: Fixed.

File: source/templates.tex (hunk 3 above)
Comment (jensmaurer) [reply source/templates.tex]: Fixed.

File: source/templates.tex [hunk 5]
```diff
@@ -3806,7 +3808,7 @@
 more specialized template is the one chosen by the partial ordering
 process.
 If both deductions succeed, the partial ordering selects
-the more constrained template as described by the rules in
+the more constrained template (if one exists) as determined below.
 \ref{temp.constr.order}.
```
[the remark this answers is not in the document]
Comment (jensmaurer) [reply source/templates.tex]: Fixed.

File: source/templates.tex (hunk 1 above)
Comment (jensmaurer) [reply source/templates.tex]: Fixed. (third level of subbullets, yay!)

**Pull request merged**